spark3-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java

```diff
@@ -0,0 +1,176 @@
+
+    // expire without retainLast param
+    List<Object[]> output1 = sql(
+        "CALL %s.system.expire_snapshots('%s', '%s', TIMESTAMP '%s')",
+        catalogName, tableIdent.namespace(), tableIdent.name(), secondSnapshotTimestamp);
+    assertEquals("Procedure output must match",
+        ImmutableList.of(row(0L, 0L, 1L)),
+        output1);
+
+    table.refresh();
+
+    Assert.assertEquals("Should expire one snapshot", 1, Iterables.size(table.snapshots()));
+
+    sql("INSERT OVERWRITE %s VALUES (3, 'c')", tableName);
+    sql("INSERT INTO TABLE %s VALUES (4, 'd')", tableName);
+    assertEquals("Should have expected rows",
+        ImmutableList.of(row(3L, "c"), row(4L, "d")),
+        sql("SELECT * FROM %s ORDER BY id", tableName));
+
+    table.refresh();
+
+    waitUntilAfter(table.currentSnapshot().timestampMillis());
+
+    Timestamp currentTimestamp = Timestamp.from(Instant.ofEpochMilli(System.currentTimeMillis()));
+
+    Assert.assertEquals("Should be 3 snapshots", 3, Iterables.size(table.snapshots()));
+
+    // expire with retainLast param
+    List<Object[]> output = sql(
+        "CALL %s.system.expire_snapshots('%s', '%s', TIMESTAMP '%s', 2)",
+        catalogName, tableIdent.namespace(), tableIdent.name(), currentTimestamp);
+    assertEquals("Procedure output must match",
+        ImmutableList.of(row(2L, 2L, 1L)),
+        output);
+  }
+
+  @Test
+  public void testExpireSnapshotUsingNamedArgs() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg", tableName);
+
+    sql("INSERT INTO TABLE %s VALUES (1, 'a')", tableName);
+    sql("INSERT INTO TABLE %s VALUES (2, 'b')", tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    Assert.assertEquals("Should be 2 snapshots", 2, Iterables.size(table.snapshots()));
+
+    waitUntilAfter(table.currentSnapshot().timestampMillis());
+
+    Timestamp currentTimestamp = Timestamp.from(Instant.ofEpochMilli(System.currentTimeMillis()));
+
+    List<Object[]> output = sql(
+        "CALL %s.system.expire_snapshots('%s', '%s', TIMESTAMP '%s', 1)",
+        catalogName, tableIdent.namespace(), tableIdent.name(), currentTimestamp);
+    assertEquals("Procedure output must match",
+        ImmutableList.of(row(0L, 0L, 1L)),
+        output);
+  }
+
+  @Test
+  public void testExpireSnapshotsGCDisabled() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg", tableName);
+
+    sql("ALTER TABLE %s SET TBLPROPERTIES ('%s' 'false')", tableName, GC_ENABLED);
+
+    AssertHelpers.assertThrows("Should reject call",
+        ValidationException.class, "Cannot expire snapshots: GC is disabled",
+        () -> sql("CALL %s.system.expire_snapshots('%s', '%s')", catalogName,
+            tableIdent.namespace(), tableIdent.name()));
+  }
+
+  @Test
+  public void testInvalidExpireSnapshotsCases() {
+    AssertHelpers.assertThrows("Should not allow mixed args",
```

Comment:
Not sure we need some of the more generic checks here, like testing if named and pos args cannot be mixed

Reply:
It is a bit weird but we have such checks in other procedures so I added for consistency.

Reply:
Sounds fine, It doesn't really matter since they run so quickly.